pmxbot/slack.py

```diff
@@ -64,14 +65,19 @@ def get_email_username_map(self, ttl_hash=get_ttl_hash()):
             }
 
     def start(self):
-        res = self.slack.rtm_connect()
-        assert res, "Error connecting"
         self.init_schedule(self.scheduler)
+        threading.Thread(target=self.run_scheduler_loop).start()
```

Comment:
I wonder if `self.slack` has an event loop that could periodically invoke the scheduler rather than creating a thread. A thread may be fine, but it could also introduce threading issues (what if a message arrives at the same time as a scheduled event).

Reply:
Oh, I see you answered the question in the description; "keep it simple". Sounds good. I've looked into the rtm API code and it uses threads to manage its concurrency, so I'm not too worried about concurrency concerns.